plugins/gcloud/lib/samson_gcloud/image_tagger.rb

```diff
@@ -11,38 +9,19 @@ def tag(deploy, output)
         return unless deploy.succeeded?
         return unless builds = deploy.project.builds.
           where(git_sha: deploy.job.commit).where.not(docker_repo_digest: nil).to_a.presence
-
-        tags = tags(deploy)
+        return unless deploy.stage.production?
```

Comment:
Could just make this one return statement?
```ruby
return if !deploy.stage.production? || deploy.stage.no_code_deployed?
```

Reply:
Maybe we should take all of these `return unless` statements L8 - L12 and put them into a method `needs_tag?`?